Mailbox/Person.cs

```diff
@@ -1,4 +1,52 @@
-﻿namespace Mailbox
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Mailbox
 {
+    public class Person : IEquatable<Person>
+    {
+        public string LastName { get; set; }
+        public string FirstName { get; set; }
+
+        public Person(string lastName, string firstName)
+        {
+            if(lastName.Equals(null) || firstName.Equals(null))
+            {
+                throw new ArgumentNullException("The name can not be null");
+            }
+
+            LastName = lastName;
+            FirstName = firstName;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is Person && Equals((Person)obj);
```

Comment:
You can use pattern matching here to avoid the cast.
```suggestion
            return obj is Person person && Equals(person);
```